pkg/operator/ceph/config/keyring/cephx.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2025 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keyring
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	v1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/version"
+)
+
+var CephAuthRotateSupportedVersion = version.CephVersion{Major: 20, Minor: 2, Extra: 0}
+
+// CephxKeyIdentifierAnnotation is the annotation that should be applied to pod specs to
+// ensure that pods restart after keys are rotated (and not restarted when keys are not rotated).
+// The keyring secret resourceVersion is suggested but not always available.
+//
+//nolint:gosec // G101: this is not hardcoded credentials
+const CephxKeyIdentifierAnnotation = "cephx-key-identifier"
```

Comment:
In addition to this annotation, could we add an annotation for the generation value? It could be a nice-to-have.

Reply:
That info is inferrable but not directly available at this point in the code with how things are set up at the moment. Can we leave this as a future item?